Add rudimentary support for hyperlinks in cells

gquipster opened the pull request:
This is rough and ready but works for me with the file I have.

I don't currently have time to add a test but will try to do so in the next few weeks.

gquipster committed:
Add rudimentary support for hyperlinks in cells

This is rough and ready but works for me with the file I have

gquipster committed:
Replace tabs with spaces

doy commented:
Let me know when you have a test for this - I'd rather not add new features that I'm not familiar with without tests (so that I don't accidentally break them in the future).

gquipster committed:
Add test XLSX file for hyperlinks

This is based upon TestActiveSheet.xls in the Spreadsheet::ParseExcel distribution

gquipster committed:
Tweaks to work with internal links as well as external links

gquipster committed:
Add test file for hyperlinks in spreadsheet

This is based upon test 47_hyperlinks.t in Spreadsheet::ParseExcel

gquipster committed:
Few tweaks to whitespace

gquipster committed:
Stop trying to be too clever

Here I was trying to match a return value in the Spreadsheet::ParseExcel test but actually lets just make this a todo test for now as otherwise I'm potentially opening up to pitfalls

gquipster committed:
Tweak to tests for hyperlinks

Per previous commit stop trying to frig values without really understanding what's going on

gquipster committed:
Yet more tweaks to whitespace

gquipster committed:
Re-write line 1

Match other test files

gquipster committed:
Reupload file to attempt to correct line endings

They are ok on my machine but the version on github seems to have \r in it

gquipster commented:
Hi,

Tweaked the code to make it slightly less rudimentary (so it now handles internal links as well as external) and to work with the same set of tests as Spreadsheet::ParseExcel.

I've marked some of the tests as TODO as I get back different values to Spreadsheet::ParseExcel - to my mind the values returned by this module are correct as they are what is in the spreadsheet, but it does differ to what Spreadsheet::ParseExcel was expecting, so I've left them expecting the values Spreadsheet::ParseExcel was for now rather than changing to what this module actually returns.

Thanks

gquipster committed:
Add twig->purge

warnes commented:
Per #97, please create a PR at https://github.com/r-gregmisc/spreadsheet-parsexlsx.
